src/Gtk/Avalonia.Cairo/Media/DrawingContext.cs

```diff
@@ -179,11 +179,25 @@ public void DrawGeometry(IBrush brush, Pen pen, Geometry geometry)
         /// <param name="rect">The rectangle bounds.</param>
         public void DrawRectangle(Pen pen, Rect rect, float cornerRadius)
         {
-            using (var p = SetPen(pen, rect.Size)) 
-            {
-                _context.Rectangle(rect.ToCairo ());
-                _context.Stroke();
-            }
+			if (cornerRadius == 0) {
+				using (var p = SetPen (pen, rect.Size)) {
+					_context.Rectangle (rect.ToCairo ());
+					_context.Stroke ();
+				}
+			} else {
+				using (var p = SetPen (pen, rect.Size)) {
+					double degrees = Math.PI / 180.0;
+					var cairoRect = rect.ToCairo ();
+
+					_context.NewSubPath ();
```

Comment:
This block for drawing a rectangle could be extracted into a method so that it is can be used in both DrawRectangle and FillRectangle